### PR TITLE
More descriptive names on GCE/GKE

### PR DESCRIPTION
**Pull request by matthewloring**
Added hostname and instanceid to debugger description to provide more
information to customers.

### CODE REVIEW

**Comment by matthewloring** [comment]
@ofrobots PTAL.

**Comment by matthewloring** [comment]
@ofrobots Ready for another look

File: config.js [hunk 1]
```diff
@@ -17,56 +17,60 @@
...
+      // To limit the size of the buffer, we truncate long strings.
+      // A value of 0 disables truncation.
+      maxStringLength: 0
+    },
 
+  // These configuration options are for internal experimentation only.
+  internal: {
+    registerDelayOnFetcherErrorSec: 300 // 5 minutes.
+  }
+}
```
Comment (ofrobots) [inline config.js]: Fix indentation.

File: index.js [hunk 2]
```diff
@@ -19,22 +19,43 @@
 // NOTE: this file is on the critical path for the startup of the user's
 // application. The path-length here needs to be minimal.
 
-var config = require('./config.js');
 var logger = require('@google/cloud-diagnostics-common').logger;
 var Debuglet = require('./lib/debuglet.js');
+var path = require('path');
+
+var initConfig = function() {
+  var util = require('util');
```
Comment (ofrobots) [inline index.js]: Move the require to the top of the file?

File: index.js [hunk 3]
```diff
@@ -19,22 +19,43 @@
...
+  var defaults = require('./config.js').debug;
+  util._extend(config, defaults);
+  if (process.env.hasOwnProperty('GCLOUD_DIAGNOSTICS_CONFIG')) {
+    var c = require(path.resolve(process.env.GCLOUD_DIAGNOSTICS_CONFIG));
+    if (c && c.debug) {
+      var capture = defaults.capture;
+      var internal = defaults.internal;
+      util._extend(capture, c.debug.capture);
+      util._extend(internal, c.debug.internal);
+      util._extend(config, c.debug);
```
Comment (ofrobots) [inline index.js]: 😷

Would using [`_.defaultsDeep`](https://lodash.com/docs#defaultsDeep) help here?

File: config.js (hunk 1 above)
Comment (matthewloring) [inline config.js]: done.

File: index.js (hunk 2 above)
Comment (matthewloring) [inline index.js]: Done.

File: index.js (hunk 3 above)
Comment (matthewloring) [inline index.js]: Much cleaner.

**Comment by ofrobots** [comment]
LGTM.

**Pull request merged**